autoprotocol/protocol.py

```diff
@@ -7998,6 +8001,10 @@ def evaporate(self, ref, mode, duration, evaporator_temperature, mode_params=Non
                         f" {evaporator_temperature}"
                     )
 
+        # remove volume from all wells in evaporated container
+        for well in ref.all_wells():
+            well.set_volume(Unit(0, "uL"))
```

Comment:
Hmm. I will say that the one aspect that I'd be concerned about here is partial evaporation. Given that evaporation has params such as `duration` and `blow_rate`, one could also probably make an assumption that, if provided with certain parameters and original volume of the container, the evaporation may not necessarily go to completion.

Instead, this is zeroing out the volume irrespective of those parameters. Arguably, what we probably want here is `volume_to_subtract = blow_rate * duration` and then make it so that per well `volume = current_volume - volume_to_subtract` and ensure that volume cannot be less than zero here.